### PR TITLE
Dry up some painless code

### PR DESCRIPTION
**Pull request by original-brownbear**
Just some Intellij automatic fixes I found while researching other things, no need to have this duplication.

### CODE REVIEW

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextDocGenerator.java [hunk 1]
```diff
@@ -742,48 +723,9 @@ private static List<PainlessContextClassInfo> sortClassInfos(
...
-                } else {
-                    compare = Boolean.compare(i1, i2) * -1;
-                }
-            }
-
-            return compare;
-        });
-
-        return classInfos;
+        classInfos.removeIf(v -> ContextGeneratorCommon.isExcludedClassInfo(v) || classExcludes.contains(v));
```
Comment (ldematte) [inline modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextDocGenerator.java:727]: Nit/question: better use a || operator or Predicate.or?

**Review by ldematte — APPROVED** [review]
LGTM, just one question (more my curiosity than anything else)

File: modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextDocGenerator.java (hunk 1 above)
Comment (original-brownbear) [reply modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextDocGenerator.java:727]: `||` is IMO the easiest to read and shortest :) Also, as far as performance is concerned it's probably faster in effectively all cases, `Predicate.or` can at best inline to the code as it is written I'd say.

**Comment by original-brownbear** [comment]
Thanks Lorenzo!

**Pull request merged**